lib/mixlib/install.rb

```diff
@@ -64,11 +64,13 @@ def available_versions
     #
     # @return [Array<String>] list of available versions for the given
     # product_name and channel.
-    def self.available_versions(product_name, channel)
+    def self.available_versions(product_name, channel, https_proxy = nil, http_proxy = nil)
```

Comment:
Perhaps it makes sense to make the 3rd parameter an opts hash? This would allow easy adding of more options in the future:

```ruby
#
# List available versions
#
# @param [String] product name
#
# @param [String, Symbol] channel
#
# @param [Hash] opts
# @option opts [String] :https_proxy 
# @option opts [String] :http_proxy 
#
# @return [Array<String>] list of available versions for the given
# product_name and channel.
def self.available_versions(product_name, channel, opts = {})
  Backend.available_versions(
    Mixlib::Install::Options.new(
      product_name: product_name,
      channel: channel.to_sym,
      https_proxy: opts[:https_proxy],
      http_proxy: opts[:http_proxy]
    )
  )
end
```